rust/chains/tw_polkadot/src/extrinsic.rs

```diff
@@ -0,0 +1,554 @@
+        let address =
+            SS58Address::from_str(&at.to_address).map_err(|_| EncodeError::InvalidAddress)?;
+        let value =
+            U256::from_little_endian_slice(&at.value).map_err(|_| EncodeError::InvalidValue)?;
+
+        let mut encoder = ExtrinsicEncoder::new().add(call_index);
+
+        // Encode asset ID if not native token
+        if at.asset_id != 0 {
+            encoder.add_mut(Compact(at.asset_id));
+        }
+
+        if !self.should_encode_raw_account() {
+            encoder.add_mut(0x00u8);
+        }
+
+        let data = encoder.add(address).add(Compact(value)).finalize();
+
+        Ok(data)
+    }
+
+    fn encode_batch(
+        &self,
+        encoded_calls: &[Vec<u8>],
+        call_indices: &Option<Proto::CallIndices>,
+    ) -> EncodeResult<Vec<u8>> {
+        let call_index = self.get_custom_call_index_or_network(UTILITY_BATCH, call_indices)?;
+        let encoder = ExtrinsicEncoder::new()
+            .add(call_index)
+            .add(Compact(encoded_calls.len()));
+
+        let data = encoded_calls
+            .iter()
+            .fold(encoder, |encoder, call| encoder.add(Raw(call)))
+            .finalize();
+
+        Ok(data)
+    }
+
+    fn encode_batch_transfer(&self, bt: &BatchTransfer) -> EncodeResult<Vec<u8>> {
+        let transfers = bt
+            .transfers
+            .iter()
+            .map(|t| self.encode_transfer(t))
+            .collect::<EncodeResult<Vec<_>>>()?;
+
+        self.encode_batch(&transfers, &bt.call_indices)
+    }
+
+    fn encode_batch_asset_transfer(&self, bat: &BatchAssetTransfer) -> EncodeResult<Vec<u8>> {
+        let transfers = bat
+            .transfers
+            .iter()
+            .map(|t| self.encode_asset_transfer(t))
+            .collect::<EncodeResult<Vec<_>>>()?;
+
+        self.encode_batch(&transfers, &bat.call_indices)
+    }
+
+    fn encode_balance_call(&self, b: &Balance) -> EncodeResult<Vec<u8>> {
+        match &b.message_oneof {
+            BalanceVariant::transfer(t) => self.encode_transfer(t),
+            BalanceVariant::batchTransfer(bt) => self.encode_batch_transfer(bt),
+            BalanceVariant::asset_transfer(at) => self.encode_asset_transfer(at),
+            BalanceVariant::batch_asset_transfer(bat) => self.encode_batch_asset_transfer(bat),
+            BalanceVariant::None => Ok(Vec::new()),
+        }
+    }
+
+    fn encode_staking_bond(&self, b: &Bond) -> EncodeResult<Vec<u8>> {
+        let call_index = self.get_custom_call_index_or_network(STAKING_BOND, &b.call_indices)?;
+        let value =
+            U256::from_little_endian_slice(&b.value).map_err(|_| EncodeError::InvalidValue)?;
+
+        let mut encoder = ExtrinsicEncoder::new().add(call_index);
+
+        if !b.controller.is_empty() {
+            // TODO: check address network ?
+            let address =
+                SS58Address::from_str(&b.controller).map_err(|_| EncodeError::InvalidAddress)?;
```

Comment:
That would be great but not required. In Cosmos we also ignored the address networks because of the complexity. It's also a responsibility for the API user to validate input addresses